guest/src/lib.rs

```diff
@@ -186,7 +186,7 @@ fn main2() -> Result<()> {
                 denominator: *ratio.denom() as u32,
             });
         println!("setting sampling rate to {:?}", value);
-        sensor.set_property(wasi::sensor::property::PropertyKey::SamplingRate, &value)?;
+        sensor.set_property(wasi::sensor::property::PropertyKey::SamplingRate, value)?;
```

Comment:
if we change here, we will get the error again when we add string, right ? If so, can we keep it ?

Reply:
Because a public API is essential to consider, even if it's not in a public repository.
It would be a problem if the API changed depending on whether it was present or not.

Reply:
> Because a public API is essential to consider, even if it's not in a public repository. It would be a problem if the API changed depending on whether it was present or not.

if you are not happy with wit-bindgen, maybe you should fix it. (or stop using it.)
also, you didn't explain why you prefer the current API over the one changed by this PR.
or, are you suggesting to add dummy fields like this to every objects?

Reply:
I just wanted to maintain consistency with Internal Repo, and I don't feel that this type of API is preferred or not.
I see, so you are saying that similar problems occur with other APIs as well. Then there's no point in just focusing on this part.